packages/core/chrome/core-chrome-browser-internal/src/ui/project/app_menu.tsx

```diff
@@ -17,26 +17,43 @@ interface AppMenuBarProps {
 }
 export const AppMenuBar = ({ headerActionMenuMounter }: AppMenuBarProps) => {
   const { euiTheme } = useEuiTheme();
+  const [height, setHeight] = React.useState<number | null>(null);
+
+  React.useLayoutEffect(() => {
+    if (height === null) return;
+    const kbnAppWrapper = document.querySelector('.kbnAppWrapper') as HTMLElement;
+    if (!kbnAppWrapper) return; // should never happen
+
+    kbnAppWrapper.style.setProperty('--kbnProjectHeaderAppActionMenuHeight', `${height}px`);
+    return () => {
+      kbnAppWrapper.style.removeProperty('--kbnProjectHeaderAppActionMenuHeight');
+    };
+  }, [height]);
+
   return (
-    <div
-      className="header__actionMenu"
-      data-test-subj="kibanaProjectHeaderActionMenu"
-      css={css`
-        z-index: ${euiTheme.levels.header};
-        background: ${euiTheme.colors.emptyShade};
-        border-bottom: ${euiTheme.border.thin};
-        display: flex;
-        justify-content: end;
-        align-items: center;
-        padding: ${euiTheme.size.s};
-        margin-bottom: -${euiTheme.border.width.thin};
-        /* fixates the elements position in the viewport, removes the element from the flow of the page */
-        position: sticky;
-        /* position below the primary fixed EuiHeader in the viewport */
-        top: var(--euiFixedHeadersOffset, 0);
-      `}
-    >
-      <HeaderActionMenu mounter={headerActionMenuMounter} />
-    </div>
+    <EuiResizeObserver onResize={(d) => setHeight(d.height)}>
```

Comment:
> We don't know the height for the action menu as it depends on the inner content and can be different

@MichaelMarcialis @tsullivan  Can you confirm if this assertion is true? I was under the impression the action menu should be a fixed height, but maybe I'm misremembering

Reply:
Here Discover (49px): 

![Screenshot 2023-10-10 at 13 18 40](https://github.com/elastic/kibana/assets/7784120/c657470c-2514-4cd7-ad0d-92f63f816351)

Here security/oblt (57px):

![Screenshot 2023-10-10 at 13 19 12](https://github.com/elastic/kibana/assets/7784120/24938509-98d8-4961-9183-10b4c9155ffe)

I agree this doesn't make sense it would be a lot more easier for us, if it'd be static